pkg/forwarder/forwarder.go

```diff
@@ -148,7 +148,12 @@ func New(cfg Config) (*Worker, error) {
 
 	// Create the `fromClient`.
 	fromTransport := metricsclient.DefaultTransport()
-	if len(cfg.FromCAFile) > 0 {
+	fromClient := &http.Client{Transport: fromTransport}
+	if cfg.Debug {
+		level.Info(logger).Log("msg", "fromClient transport: Debug ")
+		fromClient.Transport = telemeterhttp.NewDebugRoundTripper(logger, fromClient.Transport)
+	} else if len(cfg.FromCAFile) > 0 {
```

Comment:
Not really important but maybe consider using a switch over here.

Reply:
done

Reply:
Not sure I follow this. Why do we not consider the TLS artifact configured of `Debug` is set?

Reply:
Resolve in next comments